lighthouse-core/audits/resource-budget.js

```diff
@@ -0,0 +1,182 @@
+      'image': str_(i18n.UIStrings.imageResourceType),
+      'media': str_(i18n.UIStrings.mediaResourceType),
+      'font': str_(i18n.UIStrings.fontResourceType),
+      'other': str_(i18n.UIStrings.otherResourceType),
+      'third-party': str_(i18n.UIStrings.thirdPartyResourceType),
+    };
+    return strMappings[resourceType];
+  }
+
+  /**
+   * @param {Record<LH.Budget.ResourceType,ResourceEntry>} summary
+   * @return {Array<{resourceType: LH.Budget.ResourceType, label: string, requestCount: number, size: number}>}
+   */
+  static defaultTable(summary) {
+    const types = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    const rows = types.map(type => {
+      return {
+        resourceType: type,
+        label: this.rowLabel(type),
+        requestCount: summary[type].count,
+        size: summary[type].size,
+      };
+    });
+    // Force third-party to be last, descending by size otherwise
+    const thirdPartyRow = rows.find(r => r.resourceType === 'third-party') || [];
+    const otherRows = rows.filter(r => r.resourceType !== 'third-party')
+      .sort((a, b) => {
+        return b.size - a.size;
+      });
+    return otherRows.concat(thirdPartyRow);
+  }
+
+  /**
+   * @param {LH.Budget} budget
+   * @param {Record<LH.Budget.ResourceType,ResourceEntry>} summary
+   * @return {Array<{resourceType: LH.Budget.ResourceType, label: string, requestCount: number, size: number, sizeOverBudget: number | undefined, countOverBudget: string | undefined}>}
+   */
+  static budgetTable(budget, summary) {
+    const resourceTypes = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    return resourceTypes.map((type) => {
+      const resourceType = type;
+      const label = this.rowLabel(type);
+      const requestCount = summary[type].count;
+      const size = summary[type].size;
+
+      let sizeOverBudget;
+      let countOverBudget;
+
+      if (budget.resourceSizes) {
+        const sizeBudget = budget.resourceSizes.find(b => b.resourceType === type);
+        if (sizeBudget && (size > (sizeBudget.budget * 1024))) {
+          sizeOverBudget = size - (sizeBudget.budget * 1024);
+        }
+      }
+      if (budget.resourceCounts) {
+        const countBudget = budget.resourceCounts.find(b => b.resourceType === type);
+        if (countBudget && (requestCount > countBudget.budget)) {
+          const requestDifference = requestCount - countBudget.budget;
+          countOverBudget = str_(UIStrings.requestCount, {count: requestDifference});
+        }
+      }
+      return {
+        resourceType,
+        label,
+        requestCount,
+        size,
+        countOverBudget,
+        sizeOverBudget,
+      };
+    }).filter((row) => {
+      // Only resources with budgets should be included in the table
+      if (budget.resourceSizes) {
+        if (budget.resourceSizes.find(b => b.resourceType === row.resourceType)) return true;
```

Comment:
nit: `resourceSizes.some` instead of `find` (here and for `resourceCounts`) is slightly closer to what this is trying to express